File: Programmers/pro_12921_소수찾기.py
isPrime =[1]*1000001
def eratos():
    isPrime[0]=0
    isPrime[1]=0
    for i in range(2,1000000):
        if(isPrime[i]):
            j=2
            while(i*j<=1000000):
                isPrime[i*j]=0
                j+=1    
    
def solution(n):
    eratos()
    answer=0
    
    for i in range(2,n+1):
        if(isPrime[i]):
            answer += 1
    return answer

File: Programmers/pro_12921_소수찾기.js
function solution(n) {
    let answer = 0
    const isPrime = [0, 0, ...Array(n).fill(1)];

    for (let i = 2; i < n + 1; i++) {
        if (isPrime[i] == 1) {
            let j = 2
            while (i * j < n + 1)
                isPrime[i * j++] = 0
        }
    }

    for (let i = 2; i < n + 1; i++) {
        answer += isPrime[i]
    }
    return answer;
}